core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -380,10 +401,8 @@ public void transportTerminated() {
             log.warning("transportTerminated called after already terminated");
           }
           terminated = true;
+          onChannelTerminated();
```

Comment:
nit: keep ordering the same between this and shutdown(). notifyAll() and onChannelTerminated() are swapped.